From 7c9460a0a97296da279665cbcb5edec09f9ec03e Mon Sep 17 00:00:00 2001
From: Jakub Jelen <jjelen@redhat.com>
Date: Tue, 27 Oct 2026 12:09:03 +0100
Subject: [PATCH] Verify source signatures with openpgpverify

Replace the GnuPG-based %gpgverify with %openpgpverify, which uses
Sequoia sqv and takes the same options, as part of the Fedora 45 Change
https://fedoraproject.org/wiki/Changes/Sequoia_openpgpverify

Refresh B5B6416FEA6DDA05EA562A9FCB987F2783972FF9 from keyservers/WKD:
same certificate(s), with self-signatures our copy was missing; the
commands are in the spec comment.
Release bumped with a changelog entry.

Assisted-by: Claude Opus 5.5
---
 B5B6416FEA6DDA05EA562A9FCB987F2783972FF9 | 66 ++++++++++++++++++------
 rpki-client.spec                         | 13 +++--
 2 files changed, 61 insertions(+), 18 deletions(-)

diff --git a/B5B6416FEA6DDA05EA562A9FCB987F2783972FF9 b/B5B6416FEA6DDA05EA562A9FCB987F2783972FF9
index 06ab48a..b944d37 100644
--- a/B5B6416FEA6DDA05EA562A9FCB987F2783972FF9
+++ b/B5B6416FEA6DDA05EA562A9FCB987F2783972FF9
@@ -1,6 +1,4 @@
 -----BEGIN PGP PUBLIC KEY BLOCK-----
-Comment: B5B6 416F EA6D DA05 EA56  2A9F CB98 7F27 8397 2FF9
-Comment: Sebastian Benoit (software signing only) <benno@openbsd
 
 xsFNBF6U37EBEAC5tEP4ziT1w9VzzKALdevA0m0enzF3VCje9+wHWSJbdUSCQty6
 wORzimcBI5bVqbVOZHEcXmGimNKT16Ic20VCffIFeV9eoP/XOrAV8ZV3V3+R94gQ
@@ -15,17 +13,55 @@ VV2RvjjDBhbyuMllYfzPqxs9w7pSrnt6z4BoHMAOWVut6yzQM8khLLcLKY/2pFVo
 qB8hqJ1usdkhlHdbqYKYlFOS6TwpT/61IXG/uqFzUKRcloq2MBBGH6Xk6QARAQAB
 zTxTZWJhc3RpYW4gQmVub2l0IChzb2Z0d2FyZSBzaWduaW5nIG9ubHkpIDxiZW5u
 b0BvcGVuYnNkLm9yZz7CwZQEEwEIAD4CGwMFCwkIBwIGFQoJCAsCBBYCAwECHgEC
-F4AWIQS1tkFv6m3aBepWKp/LmH8ng5cv+QUCZBdXugUJCyYSiQAKCRDLmH8ng5cv
-+W4NEACOyrljdrlCRu4oUaIuxCp8dX8xbVhv72sjptrz5vfnsKO65saR8r3iyQLX
-K3R0Vlx4DseXrr7vJBBicBNdWl0i02Cxz3r4EugJ+SLfkzJesUQR9MJQIB7PjSWB
-XFLw4VH/BeuUMS1yZXdfvpTK5riGrFaNW+lLSi/ZTuNR8I3OWeG8SeLWhVFPbMlV
-evHfEWZpD5cGc7heqV3HCuVYJd9oWxI9T+To+laFA6PQ65s/sxZydSdToHvh/r0S
-5NMNBWBK29InNkuGhoe1K6NDwhJMN0a3YgkjsvopZZos8gDO0Enu++C0KTdivJYa
-g9p7xT65rKqsPc/y9V97gt7IEgA+4rpfT1PpOYVl3+fOEwdK8ynW5hjlH0/7xyqq
-jrzGsfbL0fO9MXyoXbbBlc61rz5J+6miQoykfiZyctqWQ7xsoK9CjYxtGb4wbuIP
-oSYLY962Lo7lwIRM7pu9ct/qoTlAm9EEzgZKS7I+K+Fkqa+qJwKCjflf6tjN42vZ
-Eya1d7YJzm1COGffMvS/J8poYHwvGN1mRO2R6iPIpgWJvKNa/yQeLj1RZwKGmCSe
-1jL3FL/kx3IVJc9tFUaSTZ/yYnYq5WTVyXAlDL++fBSGVB6WMk+t0HK5TBDaOAp4
-CsWpomwxECzyBwmJ9UP3nE0suls74KSlsrIsURVujkH6Keb7Dg==
-=4Gk8
+F4AWIQS1tkFv6m3aBepWKp/LmH8ng5cv+QUCad6cEwUJDwwjYgAKCRDLmH8ng5cv
++S0ZD/sFwTJTZ1Kc/5wq+ypN0CgRGqcGyFC9bRKdGlrflfsSrzgT3qAJVAIyMG9r
+n6DGT1jelw/Ntt8VxoxAhDpMYgqV00ciBwZ8iucdeGHAf1dH+miqbhB7mJ4YzgZ9
+AGIiNRq5RLyEn1+ae7z0zOmaKUWRCBwkeRiMkBCF+Ozj+I5qO3saqgPIwWmbrpLu
+YCg8JZp27tqFpzwO0kXDUEopTIcaDm6Jzv3ZFdVTTlH+vsZa4G4+aBYoD60Tiros
+CslXLdTCz2Tso3NJI4Qrd7tvvtA2OPIv/UwAKu8SP2v9f7ghJoRGeacN8ttyleZt
+pwSFVXEypSrFWhp2jd8df+Xk4QIKPrym2cHCTJ1QDuzAkIE8tb6lu6Eg4lY4VRx8
+o3HvpfE7kcQh30LU/8IAxT7p5/jJCa0n5qzKam49/cNthwoHuuKKM8JSiWefXU5o
+sSNCJBUSJAT6yDfKxIYPVKpFvggj8xg23dL+0X4N9u936jy3ebSyCdzh4JJmyAu+
+FRux8BB4oFPDMK3X5S2KdT4aBHeQfkvjLhbqZNGSGdKV0sJralyJ7n5ttRMlv6eT
+/6At94wSiCyVyhAbmGUmmG3Ao96XVi1CD2K7wqtc87yepnkNB/VE3b//CLaDL4xQ
+LX2AWNVihQnDw3SEpp/O+doey68f6OZ4O4XCV4DdKF07CgFR0MLBlAQTAQgAPgIb
+AwULCQgHAgYVCgkICwIEFgIDAQIeAQIXgBYhBLW2QW/qbdoF6lYqn8uYfyeDly/5
+BQJkF1e6BQkLJhKJAAoJEMuYfyeDly/5bg0QAI7KuWN2uUJG7ihRoi7EKnx1fzFt
+WG/vayOm2vPm9+ewo7rmxpHyveLJAtcrdHRWXHgOx5euvu8kEGJwE11aXSLTYLHP
+evgS6An5It+TMl6xRBH0wlAgHs+NJYFcUvDhUf8F65QxLXJld1++lMrmuIasVo1b
+6UtKL9lO41Hwjc5Z4bxJ4taFUU9syVV68d8RZmkPlwZzuF6pXccK5Vgl32hbEj1P
+5Oj6VoUDo9Drmz+zFnJ1J1Oge+H+vRLk0w0FYErb0ic2S4aGh7Uro0PCEkw3Rrdi
+CSOy+illmizyAM7QSe774LQpN2K8lhqD2nvFPrmsqqw9z/L1X3uC3sgSAD7iul9P
+U+k5hWXf584TB0rzKdbmGOUfT/vHKqqOvMax9svR870xfKhdtsGVzrWvPkn7qaJC
+jKR+JnJy2pZDvGygr0KNjG0ZvjBu4g+hJgtj3rYujuXAhEzum71y3+qhOUCb0QTO
+BkpLsj4r4WSpr6onAoKN+V/q2M3ja9kTJrV3tgnObUI4Z98y9L8nymhgfC8Y3WZE
+7ZHqI8imBYm8o1r/JB4uPVFnAoaYJJ7WMvcUv+THchUlz20VRpJNn/JidirlZNXJ
+cCUMv758FIZUHpYyT63QcrlMENo4CngKxamibDEQLPIHCYn1Q/ecTSy6WzvgpKWy
+sixRFW6OQfop5vsOwsGUBBMBCAA+AhsDBQsJCAcCBhUKCQgLAgQWAgMBAh4BAheA
+FiEEtbZBb+pt2gXqViqfy5h/J4OXL/kFAmB4mAoFCQWmH1kACgkQy5h/J4OXL/mp
+Uw/+PLJZzCedk2tuMo6kaItngZ49jgjtFiXJ+8dslx5kG6HlD7m73hb+R71g4XnV
+bwMH/Y5PckJxKCpMu9SjZYd8eYpVvlzOc0K2R84CwIAK1R81J8eMDAIvdk7ggPoj
+EZy94vbBNvANp1BrugpOpkMJDnQ9YQ/m2vFAPKkFQKFfgIIOwjZJrgHz39j/K6vB
+6PS5+GVsBX6roCefNN6IiumYByG/PB8e0ZJSw1xuiRtgSDHksEg8t1qXNyhHxajr
+jUrWrxm4RK4tD1hInxk2HB3NHKkXHPd05/dcJUq3FUW0QiDysNN5MSUD2VdkCXpI
+MiRioXBIxS4gxWP4m8jaUnTagT8pwTgX3GNdZfle3R+PT2isd+2mMKy00B3L0HrI
+aY8VFubgjIP+LPqnf0K7XQLjKf19zKhits0ulo1H7xLyka2wDn3/Iiyb+v4nrrx1
+QB4Wg+NREmBJ90uaWq1BWrOx1uLUUEF30Lab9xyMNTt6y8eJX+wan10bQegZSUBo
+HqKb/fng7VwJAHXWcowAKzameZCZWbQt//R/FCKmABko5/jSuOrgeD/iNiP0vhsy
+NUt8gbMPtRAPBY2EE8os9xUqACEoJC6C9Q//a+M0+WR+Z0ACBqVyIVL8nSUjWxys
+LR4ioOm6nkKRjyKXHvp/btZ7hB05yLfPLNKpuj//5JIUdkbCwZQEEwEIAD4WIQS1
+tkFv6m3aBepWKp/LmH8ng5cv+QUCXpTfsQIbAwUJAeEzgAULCQgHAgYVCgkICwIE
+FgIDAQIeAQIXgAAKCRDLmH8ng5cv+TUREACocx0oSH0C5IMMHbayWuSXKqF4OppH
+qGdaWGuvSYa/ERjHHPYRgsp8pBJp8eb/JpNJ5PiXa0Li6HRSTjDTbWRccMI4FEUR
+974pU6sq7sQA2vg85tVb5dC5R4H50uYv4hMarX5tMx3h0/C2+u+JG+d+VCmTDd3a
+6bOCXWZtD0nEejP1u8Lv7Iad0VL0aXv0LUy4cdkwJKE6tdjEbPYgwWYtVC9bukZf
+SJixEGXwXF3eAZyVeQ+M4taasUDIqptfQN6061TlIcZanQhBd199nJjAiiSTf/pO
+SyfAxfgwybr4ZrtwqsGxC7geEjW79st3XD0AlYkUBwxsKf2RbZhdOfvcOe5te8kw
+sX4d/xoK2hs063tFV3sgmaSdpCz1uwBnDk7mjHj1/RCct7eAWrVXvgnZgwkflJAv
+jest0rvBgxjwH7YAoBcRvBxbcXKP7YCVuLZcanUFn2mTYyGhRE8TO+LsLqgt45jl
+yuzFsWuRa5hn+j+bJuWObMyGZkWlOxOh2+L3ISuGgGEjZV1PCDrwlNNADTcRCMTu
+x2An7o6xknosRvoetJsa4yxrchQVCexvaEyQrBPKViFXzRhaaTX7n+SxuPxsbHRX
+Ddl1pYF/3AsH5yBStj30pQnHTr//lIapTNe/+gU//cl0w1JcBoggdZ0T1NMiGaJf
+mb4Qw1CW1siLIg==
+=5r+8
 -----END PGP PUBLIC KEY BLOCK-----
diff --git a/rpki-client.spec b/rpki-client.spec
index 954c9d5..2b1c25d 100644
--- a/rpki-client.spec
+++ b/rpki-client.spec
@@ -9,7 +9,7 @@
 Summary:        OpenBSD RPKI validator to support BGP Origin Validation
 Name:           rpki-client
 Version:        9.9
-Release:        1%{?with_snapshot:.git%{gitdate}}%{?dist}
+Release:        2%{?with_snapshot:.git%{gitdate}}%{?dist}
 # rpki-client itself is ISC but uses other source codes, breakdown:
 # BSD-2-Clause: include/sys/tree.h and src/{http,output}.c
 # BSD-3-Clause: compat/{setproctitle,vis}.c and include/{sha2_openbsd,vis,sys/queue}.h and src/mkdir.c
@@ -21,6 +21,10 @@ URL:            https://www.rpki-client.org/
 %if !0%{?with_snapshot}
 Source0:        https://ftp.openbsd.org/pub/OpenBSD/rpki-client/%{name}-%{version}.tar.gz
 Source1:        https://ftp.openbsd.org/pub/OpenBSD/rpki-client/%{name}-%{version}.tar.gz.asc
+# Refreshed from keyservers/WKD (same keys, newer signatures):
+#   sq --home=none network search --output - B5B6416FEA6DDA05EA562A9FCB987F2783972FF9 | sq keyring filter --experimental --cert B5B6416FEA6DDA05EA562A9FCB987F2783972FF9 --output update.pgp
+#   sq keyring merge --output B5B6416FEA6DDA05EA562A9FCB987F2783972FF9.new B5B6416FEA6DDA05EA562A9FCB987F2783972FF9 update.pgp
+#   mv B5B6416FEA6DDA05EA562A9FCB987F2783972FF9.new B5B6416FEA6DDA05EA562A9FCB987F2783972FF9
 Source2:        https://keys.openpgp.org/vks/v1/by-fingerprint/B5B6416FEA6DDA05EA562A9FCB987F2783972FF9
 %else
 Source0:        https://github.com/rpki-client/rpki-client-portable/archive/%{portable_commit}/%{name}-portable-%{version}-%{portable_shortcommit}.tar.gz
@@ -31,7 +35,6 @@ Source4:        %{name}.service
 Source5:        %{name}.timer
 Source6:        %{name}.service.el8
 %if !0%{?with_snapshot}
-BuildRequires:  gnupg2
 %else
 BuildRequires:  autoconf
 BuildRequires:  automake
@@ -45,6 +48,7 @@ BuildRequires:  expat-devel
 BuildRequires:  rsync
 BuildRequires:  systemd-rpm-macros
 BuildRequires:  zlib-devel
+BuildRequires:  openpgpverify
 Requires:       rsync
 %{?systemd_requires}
 %{?sysusers_requires_compat}
@@ -62,7 +66,7 @@ also as CSV or JSON objects for consumption by other routing stacks.
 
 %prep
 %if !0%{?with_snapshot}
-%{gpgverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
+%{openpgpverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
 %setup -q
 %else
 %setup -q -n %{name}-portable-%{portable_commit}
@@ -111,6 +115,9 @@ install -D -p -m 0644 %{SOURCE5} $RPM_BUILD_ROOT%{_unitdir}/%{name}.timer
 %dir %attr(0755,%{name},%{name}) %{_localstatedir}/lib/%{name}/
 
 %changelog
+* Fri Oct 23 2026 Jakub Jelen <jjelen@redhat.com> - 9.9-2
+- Verify upstream source signatures with openpgpverify
+
 * Mon Aug 24 2026 Robert Scheck <robert@fedoraproject.org> 9.9-1
 - Upgrade to 9.9 (#2521687)
 
-- 
2.56.0

